src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/Context/FormattingContext.IndentationData.cs

```diff
@@ -42,25 +57,34 @@ public SimpleIndentationData(TextSpan textSpan, int indentation)
             }
 
             public override int Indentation => _indentation;
+
+            protected override IndentationData WithTextSpanCore(TextSpan span)
+            {
+                return new SimpleIndentationData(span, _indentation);
+            }
         }
 
-        private class LazyIndentationData : IndentationData
+        private class RelativeIndentationData : IndentationData
         {
-            private readonly Lazy<int> _indentationGetter;
-            public LazyIndentationData(TextSpan textSpan, Lazy<int> indentationGetter)
+            private readonly Lazy<int> _lazyIndentationDelta;
```

Comment:
so the idea is htat we wait until we actually need this, then comput it, then cache it at that point?

Reply:
That was the idea even before this PR. The problem stemmed from the fact that _too much_ of the computed result was captured in that lazily-computed value. The part that changes is now a separate lambda which is called every time instead of cached.